dist/exceptionless.d.ts

```diff
@@ -31,6 +31,10 @@ export interface IEnvironmentInfoCollector {
 export interface IErrorParser {
     parse(context: EventPluginContext, exception: Error): IError;
 }
+export interface IExitController {
```

Comment:
Not sure about the word controller here. Any particular reason you chose that? I am thinking maybe just IExitHandler.  Also, I'm not sure if this needs to be in the core client. I would think that each plugin would just do whatever it needs to do like wire up to the app exiting and call the client's process queue method.

I just looked through the code some more and see that you are using this to trigger a sync queue submission. What if we just added a IEventQueue.processSync method that could be called?  Then from your node client you could listen to the exit event and call the processSync method. We could do the same thing in the browser when the browser window is closing.

I think we would need to have a ISubmissionClient.postEventsSync as well.  But then this is a generic behaviour in the client and any plugins can just listen to whatever events they want to listen for and then decide what to do. In the exit events then they would call the sync version to make sure all the events are sent before exiting.

What do you think @frankebersoll ?